models/taskRequests.js

```diff
@@ -0,0 +1,127 @@
+const { TASK_REQUEST_STATUS } = require("../constants/taskRequests");
+const { TASK_STATUS } = require("../constants/tasks");
+const { USER_STATUS } = require("../constants/users");
+const firestore = require("../utils/firestore");
+const { toFirestoreData } = require("../utils/taskRequests");
+const usersCollection = firestore.collection("users");
+const taskRequestsCollection = firestore.collection("taskRequests");
+const tasksCollection = firestore.collection("tasks");
+const tasksModel = require("./tasks");
+
+/**
+ * Fetch all task requests
+ *
+ * @return {Object}
+ */
+const fetchTaskRequests = async () => {
+  try {
+    const taskRequests = await taskRequestsCollection.get();
+
+    const taskRequestsData = [];
+
+    taskRequests.forEach((taskRequest) => taskRequestsData.push({ id: taskRequest.id, ...taskRequest.data() }));
+
+    return taskRequestsData;
+  } catch (err) {
+    logger.error("error fetching tasks", err);
+    throw err;
+  }
+};
+
+/**
+ * Creates a task request
+ *
+ * @param taskId { string }: id of task request
+ * @param userId { string }: userId of user whose creating the task request
+ * @return {Promise<{taskRequest: Object}>}
+ */
+const createTaskRequest = async (taskId, userId) => {
```

Comment:
can we rename it to createOrUpdateRequest

Reply:
+1